kythe/cxx/indexer/textproto/analyzer.cc

```diff
@@ -217,6 +236,120 @@ Status TextprotoAnalyzer::AnalyzeMessage(
   return OkStatus();
 }
 
+// Given a type url that looks like "type.googleapis.com/example.Message1",
+// returns "example.Message1".
+std::string ProtoMessageNameFromAnyTypeUrl(absl::string_view type_url) {
+  std::vector<std::string> parts = absl::StrSplit(type_url, "/");
+  if (parts.size() != 2) {
+    LOG(ERROR) << "Invalid Any type_url: '" << type_url << "'";
```

Comment:
This will end up failing for Kythe type urls (which is our own fault, but still).  I would do something like:
// Return the substring from after the last '/' to the end or an empty string.
// If there is no slash, returns the entire string.
return std::string(type_url.substr(std::min(type_url.size(), type_url.rfind('/') + 1)));

Reply:
I tried changing the type url in one of my test cases to "kythe.io/proto/any_type_test.Message2" and it fails to parse:
`Error parsing text-format any_type_test.Message: 28:22: Expected "]", found "/"`.

It looks like the url we're using isn't actually legal textproto?

Reply:
(that error's from TextFormat::Parser, not the analyzer)

Reply:
It technically is, but also frequently fails to work with various combinations of language and format (it works from Go, for example, but fails without a custom TypeResolver in C++ when parsing a json proto, etc.).  In practice, it was a poor choice but technically fine.  Still, we should try to support it in our tools when possible ;-)

Reply:
Sorry, "it technically is" should be "technically is fine".  As documented, any scheme-less URL should be fine there, including one that has more than a single slash, it's just that the C++ parser fails to follow the spec :-(

Reply:
Works for me. I swapped in your code snippet to take everything after the last slash.

Reply:
I'll need to adjust my regex to handle this too.

Reply:
Actually since I'm using TextFormat::Parser and it fails to parse with extra slashes, I can't really support it anyways :/